cmake/CMakeLists.txt

```diff
@@ -137,6 +137,7 @@ option(onnxruntime_FUZZ_TEST "Enable Fuzz testing" OFF)
 option(onnxruntime_ENABLE_NVTX_PROFILE "Enable NVTX profile." OFF)
 option(onnxruntime_ENABLE_MEMORY_PROFILE "Enable memory profile." OFF)
 option(onnxruntime_ENABLE_TRAINING "Enable training functionality." OFF)
+option(onnxruntime_ENABLE_TRAINING_API "Enable training C++ apis." OFF)
```

Comment:
change wording to make it specific to on device training? We have other c++ apis as well for training and this generic name will be confusing. Later when we create a on device training specific build we can use the same cmake option

Reply:
Make sense, adapted the name.